sgkit/tests/io/vcf/test_vcf_reader.py

```diff
@@ -1528,5 +1531,21 @@ def test_vcf_to_zarr__retain_files(shared_datadir, tmp_path, retain_temp_files):
         target_part_size="500B",
     )
     ds = xr.open_zarr(output)
-    assert ds.attrs["contigs"] == ["19", "20", "X"]
+    assert_array_equal(ds["contig_id"], ["19", "20", "X"])
     assert (len(os.listdir(temp_path)) == 0) != retain_temp_files
+
+
+def test_vcf_to_zarr__legacy_contig_and_filter_attrs(shared_datadir, tmp_path):
+    path = path_for_test(shared_datadir, "sample.vcf.gz")
+    output = tmp_path.joinpath("vcf.zarr").as_posix()
+
+    vcf_to_zarr(path, output, chunk_length=5, chunk_width=2)
+    ds = xr.open_zarr(output)
+
+    # drop new contig_id and filter_id variables
+    ds = ds.drop_vars(["contig_id", "filter_id"])
+
+    # check that contigs and filters can still be retrieved
+    assert num_contigs(ds) == 3
+    assert_array_equal(get_contigs(ds), np.array(["19", "20", "X"], dtype="S"))
```

Comment:
Could use `pytest.warns` here to check the warning.

Reply:
Fixed